### PR TITLE
Fixes ooc bans and latejoin antag jobbans

### LINKED ISSUE #16423: OOC Bans apparently don't do anything.
I've been informed by the admin team that OOC bans don't actually prevent people from speaking in OOC chat.

plz fix

### PR DESCRIPTION
**Pull request by Shadowlight213**
fixes #16423

### CODE REVIEW

**Comment by duncathan** [comment]
why does this fix it?

**Comment by Shadowlight213** [comment]
@duncathan because jobban_isbanned takes a mob as an argument, not a client.
/proc/jobban_isbanned(mob/M, rank)
tkdrg must have changed it
looking at it, it looks like there's a number of other stuff that's still sending a client. FUN

**Comment by KorPhaeron** [comment]
> because jobban_isbanned takes a mob as an argument, not a client.

??

**Comment by Shadowlight213** [comment]
@KorPhaeron the ooc ban was giving the proc a client instead of a mob.
the argument is typecasted to mob, so it wasn't receiving anything to check against.

**Comment by KorPhaeron** [comment]
I mean why would you check a ban against a mob instead of a client in the first place

**Comment by Shadowlight213** [comment]
@KorPhaeron ask tkdrg. they messed with it.

**Comment by KorPhaeron** [comment]
He's dead though

**Comment by Shadowlight213** [comment]
Get a ouija board. also merge please.

File: code/game/verbs/ooc.dm [hunk 1]
```diff
@@ -30,9 +30,10 @@
 		if(prefs.muted & MUTE_OOC)
 			src << "<span class='danger'>You cannot use OOC (muted).</span>"
 			return
-		if(jobban_isbanned(src, "OOC"))
-			src << "<span class='danger'>You have been banned from OOC.</span>"
-			return
+		if(src.mob)
+			if(jobban_isbanned(src.mob, "OOC"))
```
Comment (optimumtact) [inline code/game/verbs/ooc.dm]: this would have been nicer is if(src.mob && jobbanned_isbanned(blah))

**Pull request merged**